dbscripts/xml/upgrade/3.2.0_versioning.xml

```diff
@@ -143,6 +143,19 @@
 			)
 		</query>
 
+		<!--
+			Copy over updated submission_galleys table with supplementary files as galleys.
+			These are otherwise lost as these changes are added to submission_galleys but never
+			copied over to temp_submission_galleys.
+		-->
+
+		<!-- First remove temp table created in 3.2.0_preupdate_versioning -->
+		<drop table="temp_submission_galleys" />
```

Comment:
Are you sure this table won't be necessary for the 3.1.x to 3.2.x upgrade path? Temporary tables are generally created no earlier than necessary and removed no later.

Reply:
I had wondered about that as well, but I didn't see it being used anywhere else. It was created in `3.2.0_preupdate_versioning.xml`, moved from the temp table to the publication_galleys table and subsequently dropped in `3.2.0_versioning.xml`. Are there any other ways it might affect the 3.1.x to 3.2.x upgrade process other than writing to `temp_submission_galleys`?

Reply:
Sorry for the delay in responding --

In summary, the `temp_submission_galleys` table is used as follows as part of the 3.1.x to 3.2.x introduction of metadata versioning:

1. Backup the `submission_galleys` table contents in `temp_submission_galleys`.
2. (do other 2.x upgrade things)
3. (do other 3.x upgrade things)
4. Adapt the content in `temp_submission_galleys` into `publication_galleys`.

The problem you're encountering with the 2.x to 3.x upgrade, I think, is that step 2 includes population of the `submission_galleys` rather than `temp_submission_galleys`, and that new content isn't handled in step 4.

I think what your PR is doing changes it to...

1. Backup the `submission_galleys` table contents in `temp_submission_galleys`.
2. (do other 2.x upgrade things affecting `submission_galleys`)
3. (do other 3.x upgrade things affecting `temp_submission_galleys`)
4. Throw away and re-create `temp_submission_galleys`
5. Adapt the content in `temp_submission_galleys` into `publication_galleys`.

This will suffer the reverse problem -- the 2.x upgrade might be improved but the 3.1.x to 3.2.x upgrade will suffer.

I think the solution you're looking for might be to find cases where the 2.x to 3.x upgrade process refers to `submission_galleys` after `temp_submission_galleys` has been created, and change them to use `temp_submission_galleys` instead.

Reply:
Thanks! Yes, that's exactly right. I'll take a look at re-addressing the 2.x to 3.x upgrade issue in the way you mentioned.

Reply:
ye